AsyncDisplayKit/ASImageNode.mm

```diff
@@ -273,35 +273,24 @@ - (UIImage *)displayWithParameters:(id<NSObject> *)parameter isCancelled:(asdisp
     return nil;
   }
   
-  CGRect drawParameterBounds    = CGRectZero;
-  BOOL forceUpscaling           = NO;
-  CGSize forcedSize             = CGSizeZero;
-  BOOL cropEnabled              = YES;
-  BOOL isOpaque                 = NO;
-  UIColor *backgroundColor      = nil;
-  UIViewContentMode contentMode = UIViewContentModeScaleAspectFill;
-  CGFloat contentsScale         = 0.0;
-  CGRect cropDisplayBounds      = CGRectZero;
-  CGRect cropRect               = CGRectZero;
-  asimagenode_modification_block_t imageModificationBlock;
-
+  ASImageNodeDrawParameters drawParameter;
   {
     ASDN::MutexLocker l(__instanceLock__);
```

Comment:
Nice improvement overall!  I suggest one small further improvement, for a tiny perf win and a nice readability win:

__instanceLock__.lock();
[Indent two spaces] ASImageNodeDrawParameters drawParameters = _drawParameters;
__instanceLock__.unlock();

This way the variable declaration can be on the same line as the assignment, no C++ stack object, and a clearer reading of "unlock before proceeding"